app/uk/gov/hmrc/exports/repositories/TransactionalOps.scala

```diff
@@ -50,37 +51,43 @@ class TransactionalOps @Inject()(
         None
     }
 
-  private def addSummariesToSubmissionAndUpdate(
+  private def addNotificationSummariesToSubmissionAndUpdate(
     session: ClientSession,
     actionId: String,
     notifications: Seq[ParsedNotification],
     submission: Submission
   ): Future[Option[Submission]] = {
     val index = submission.actions.indexWhere(_.id == actionId)
     val action = submission.actions(index)
-
     val seed = action.notifications.fold(Seq.empty[NotificationSummary])(identity)
 
-    def prependNotificationSummary(accumulator: Seq[NotificationSummary], notification: ParsedNotification): Seq[NotificationSummary] =
-      NotificationSummary(notification, submission.actions, accumulator) +: accumulator
-
-    // Parsed notifications need to be sorted (asc), by dateTimeIssued, due to the (ACCEPTED => GOODS_ARRIVED_MESSAGE) condition
-    val notificationSummaries = notifications.sorted.foldLeft(seed)(prependNotificationSummary)
+    val (actionWithAllNotificationSummaries, notificationSummaries) =
+      updateActionWithNotificationSummaries(action, submission, notifications, seed)
 
-    // The resulting notificationSummaries are sorted again, this time (dsc), since it's not sure
-    // if they are more recent or not of the (maybe) existing notificationSummaries of the action.
-    val actionWithAllNotificationSummaries = action.copy(notifications = Some(notificationSummaries.sorted.reverse))
+    if (action.requestType == CancellationRequest)
```

Comment:
Nit: In future we will also have `AmendRequest` action types (that should behave similar to the `CancellationRequest` type in not updating the `latestEnhancedStatus`). Just thinking it would be better to swap this `if` logic to be `if (action.requestType == SubmissionRequest)` to support this future addition (although we can always change it later on it would be one less thing to remember to change) :slightly_smiling_face: